beefy-primitives/src/commitment.rs

```diff
@@ -95,6 +96,130 @@ impl<TBlockNumber, TPayload, TSignature> SignedCommitment<TBlockNumber, TPayload
 	}
 }
 
+/// Type to be used to denote placement of signatures
+type BitField = Vec<u8>;
+/// Compress 8 bit values into a single u8 Byte
+const CONTAINER_BIT_SIZE: usize = 8;
+
+/// Temporary representation used for encoding efficiency.
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode)]
+struct TemporarySignatures<TCommitment, TSignature> {
+	commitment: TCommitment,
+	signatures_from: BitField,
+	signatures_no: u32,
+	signatures: Vec<TSignature>,
+}
+
+impl<'a, TBlockNumber, TPayload, TSignature> From<&'a SignedCommitment<TBlockNumber, TPayload, TSignature>>
+	for TemporarySignatures<&'a Commitment<TBlockNumber, TPayload>, &'a TSignature>
+where
+	TSignature: Encode,
+	TBlockNumber: Encode,
+	TPayload: Encode,
+{
+	/// Convert `SignedCommitment`s into `TemporarySignatures` that are packed better for
+	/// network transport.
+	fn from(signed_commitment: &'a SignedCommitment<TBlockNumber, TPayload, TSignature>) -> Self {
```

Comment:
I know that I suggested `From` implementations initially, but I now things that it might be better of as regular `imlpl` blocks. Also I think the bounds are not really necessary at this point, are they?:


```rust
impl<'a, TBlockNumber, TPayload, TSignature>
  TemporarySignatures<&'a Commitment<TBlockNumber, TPayload>, &'a TSignature>
{
  fn pack(signed_commitment: &'a SignedCommitment<TBlockNumber, TPayload, TSignature>) -> Self {
    ...
  }
}

impl<'a, TBlockNumber, TPayload, TSignature>
  TemporarySignatures<Commitment<TBlockNumber, TPayload>, TSignature>
{
  fn unpack(self) -> SignedCommitment<TBlockNumber, TPayload, TSignature> {
    ...
  }
}
```

Reply:
We could do it the way illustrated in your code, removing the bounds doesn't seem to work though...

Reply:
Have you tried removing the bounds only for `pack/unpack` methods? They are definitely needed for `Encode/Decode` implementation though.

Reply:
Yes, the wierd compile errors start popping up, let me try this again when I get the time over the weekend.

Reply:
Make sure to follow the [Build guide](https://github.com/paritytech/grandpa-bridge-gadget/blob/master/README.md#build) from readme.

Reply:
I have tried a few stuff, but the same issues that plagued me earlier with lifetime annotation keep coming to haunt here... Could you take a look? If not, seems to be resolved.

Reply:
This was a big misconception on my side, seems like I got it figured, haha